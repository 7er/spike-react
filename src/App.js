import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';


import VisibleTodoList from './containers/VisibleTodoList'


export class App extends Component {
  render() {
    return (
        <VisibleTodoList />
    );
  }
}

export default App;
