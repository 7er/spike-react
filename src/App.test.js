import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { createStore } from 'redux'
import { Provider } from 'react-redux'


import App from './App';
import todoApp from './reducers'
import {
  addTodo,
  toggleTodo,
  setVisibilityFilter,
    VisibilityFilters
} from './actions'



const configStore = () => {
    const store = createStore(todoApp)
    store.dispatch(addTodo('Learn about react'))
    store.dispatch(addTodo('Learn about reducers'))
    store.dispatch(addTodo('Learn about store'))
    store.dispatch(toggleTodo(0))
    store.dispatch(toggleTodo(1))
    store.dispatch(setVisibilityFilter(VisibilityFilters.SHOW_COMPLETED))
    return store;
}

it('renders without crashing', () => {
    const store = configStore();
    const div = document.createElement('div');
    ReactDOM.render(<Provider store={store}><App /></Provider>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('renders the TodoList correctly', () => {
    const store = configStore();
    const component = renderer.create(<Provider store={store}><App /></Provider>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
