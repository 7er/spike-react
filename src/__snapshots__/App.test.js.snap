// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the TodoList correctly 1`] = `
<ul>
  <li>
    <strike>
      Learn about react
    </strike>
     
  </li>
  <li>
    <strike>
      Learn about reducers
    </strike>
     
  </li>
</ul>
`;
